sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -68,6 +70,14 @@ class SimpleFunctionRegistry extends FunctionRegistry {
   override def lookupFunction(name: String): Option[ExpressionInfo] = {
     functionBuilders.get(name).map(_._1)
   }
+
+  override def copy(): SimpleFunctionRegistry = {
```

Comment:
SimpleFunctionRegistry is not designed to be used by multiple threads (read/write in the same time), I think it's fine now, because the builtin is constructed in the begging.